### DISCUSSION TITLE
Set metadata for existing file

### DISCUSSION THREAD

**Discussion by BorisWilhelms** [SixLabors/ImageSharp#1427] [Q&A] [answered]
I am trying to update EXIF information for a bunch of file. Unfortunately some of the files to not have metadata (Metadata property is null). How can I add a metadata object to an existing file, if it is null. The property itself is readonly (get-only). 

Example code: 

```csharp
                    using var image = Image.Load(file.FullName);
                    if (image.Metadata == null)
                    {
                        Console.WriteLine($"{file.Name}: No metadata found.");
                        return;
                    }
```

**Comment by brianpopow** [comment, marked as answer]
The Metadata property should always be != null. It is possible that the Exif field of metadata has no value, if the image has no Exif profile.

- Which version of ImageSharp are you using?
- Can you provide the image where you see this?

**Reply by BorisWilhelms** [reply]
Seems like you are right. And I guess the issue was on the Exif-profile property. I can not reproduce it. Sorry for the confusion!
